pkg/providers/tinkerbell/upgrade.go

```diff
@@ -209,6 +207,45 @@ func (p *Provider) RunPostControlPlaneUpgrade(ctx context.Context, oldClusterSpe
 	return nil
 }
 
+func (p *Provider) ValidateNewSpec(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error {
+	prevSpec, err := p.providerKubectlClient.GetEksaCluster(ctx, cluster, clusterSpec.Cluster.Name)
+	if err != nil {
+		return err
+	}
+
+	prevDatacenterConfig, err := p.providerKubectlClient.GetEksaTinkerbellDatacenterConfig(ctx, prevSpec.Spec.DatacenterRef.Name, cluster.KubeconfigFile, prevSpec.Namespace)
+	if err != nil {
+		return err
+	}
+
+	datacenterConfig := p.datacenterConfig
+
+	oSpec := prevDatacenterConfig.Spec
+	nSpec := datacenterConfig.Spec
```

Comment:
nit: Can we directly do `p.datacenterConfig.Spec`? I see `datacenter` is not used anywhere else.